src/main/java/com/android/volley/toolbox/DiskBasedCacheUtility.java

```diff
@@ -37,6 +43,95 @@ static String getFilenameForKey(String key) {
         return localFilename;
     }
 
+    /** Returns a file object for the given cache key. */
+    public static File getFileForKey(String key, FileSupplier rootDirectorySupplier) {
+        return new File(rootDirectorySupplier.get(), getFilenameForKey(key));
+    }
+
+    static boolean checkPrune(long totalSize, int dataLength, int maxCacheSize) {
```

Comment:
You could make the function more versatile and really ensure all the checks are in sync by having it take a `newTotalSize` instead of both a `totalSize` and `dataLength`